### PR TITLE
[Core][GeometricalObject] Add IsActive

### PR DESCRIPTION
**Pull request by philbucher**
**📝 Description**
Adding `IsActive` to `GeometricalObject` (and hence to `Element` and `Condition`)

I think this makes sense since this is duplicated all over the place (12 times only in the Block B&S)

I will add tests if this gets accepted

FYI @KratosMultiphysics/geomechanics I think you use this quite a lot

### CODE REVIEW

**Comment by RiccardoRossi** [comment]
honestly we should remove the need of "isdefined" (although admittedly it is my bad ot is there).

that way flags would occupy one double instead of 2

File: kratos/sources/geometrical_object.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/geometrical_object.h"
+#include "includes/kratos_flags.h"
+
+namespace Kratos {
+
```
Comment (loumalouomega) [inline kratos/sources/geometrical_object.cpp:22]: I would move more things to the cpp to reduce compilation times

**Review by loumalouomega — COMMENTED** [review]
Okay from my side

**Comment by RiccardoRossi** [comment]
i would also be ok with  this. @KratosMultiphysics/technical-committee?

**Comment by rubenzorrilla** [comment]
> i would also be ok with this. @KratosMultiphysics/technical-committee?

Makes also sense for me.

**Comment by philbucher** [comment]
Pung @roigcarlo @sunethwarna @pooyan-dadvand

File: kratos/sources/geometrical_object.cpp (hunk 1 above)
Comment (pooyan-dadvand) [reply kratos/sources/geometrical_object.cpp:22]: But in a separate PR. It is important to keep the code reorganization PR separate from the otheres.

**Comment by pooyan-dadvand** [comment]
BTW, why we need to check if is defined?

**Comment by RiccardoRossi** [comment]
if it is not defined it would default to false...

**Comment by philbucher** [comment]
Any more comments?
I would like to use this in some refactorings :)

File: kratos/sources/geometrical_object.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+bool GeometricalObject::IsActive() const 
+{
+    return IsDefined(ACTIVE) ? Is(ACTIVE) : true;
```
Comment (pooyan-dadvand) [inline kratos/sources/geometrical_object.cpp:25]: In checking for true, the `IsDefined` call is redundant. The `IsDefined` shall be use if we want to check for false in which we don't know if the flag is unset or is really set to false.

Comment (philbucher) [reply kratos/sources/geometrical_object.cpp:25]: I am sorry I am a bit lost :sweat_smile: 
Can you please post your suggestion?

Comment (pooyan-dadvand) [reply kratos/sources/geometrical_object.cpp:25]: It is correct! Sorry for the noise...

**Review by pooyan-dadvand — APPROVED** [review]
I am approving as @KratosMultiphysics/technical-committee members already gave their thumbs ups!

**Pull request merged**